nomad/fsm.go

```diff
@@ -1501,7 +1525,14 @@ func (n *nomadFSM) Restore(old io.ReadCloser) error {
 			if err := restore.CSIVolumeRestore(plugin); err != nil {
 				return err
 			}
-
+		case EventSnapshot:
+			event := new(structs.Events)
+			if err := dec.Decode(event); err != nil {
+				return err
+			}
+			if err := restore.EventRestore(event); err != nil {
+				return err
+			}
```

Comment:
If the flag is is false, should we ignore `EventSnapshot` events as well when `n.config.EnableEventBroker`?